src/components/Table.jsx

```diff
@@ -0,0 +1,238 @@
+import cn from 'classnames';
+// import { useState } from 'react';
+
+const QUERY = {
+  ID: 'ID',
+  PRODUCT: 'PRODUCT',
+  CATEGORY: 'CATEGORY',
+  USER: 'USER',
+};
+
+const sorter = (products, query) => {
+  const sortedProducts = structuredClone(products);
+
+  if (!Object.keys(query).length) {
+    return sortedProducts;
+  }
+
+  const [property] = Object.entries(query);
+  const [key, value] = property;
+
+  switch (key) {
+    case QUERY.ID:
+      sortedProducts.sort((a, b) => (
+        value === 'ASC'
+          ? a.id - b.id
+          : b.id - a.id
+      ));
+      break;
+
+    case QUERY.USER:
+      sortedProducts.sort((a, b) => (
+        value === 'ASC'
+          ? a.user.name.localeCompare(b.user.name)
+          : b.user.name.localeCompare(a.user.name)
+      ));
+      break;
+
+    case QUERY.CATEGORY:
+      sortedProducts.sort((a, b) => (
+        value === 'ASC'
+          ? a.category.title.localeCompare(b.category.title)
+          : b.category.title.localeCompare(a.category.title)
+      ));
+      break;
+
+    case QUERY.PRODUCT:
+      sortedProducts.sort((a, b) => (
+        value === 'ASC'
+          ? a.name.localeCompare(b.name)
+          : b.name.localeCompare(a.name)
+      ));
+      break;
+
+    default:
+      return sortedProducts;
+  }
+
+  return sortedProducts;
+};
+
+export const Table = ({ products, query, setQuery }) => {
+  // const [query, setQuery] = useState({});
+
+  const sortedProducts = sorter(products, query);
```

Comment:
I think `Table` component should only be responsible for outputting the table, the rest of the logic like filtering, sorting, etc. should happen in parent components. This way you can reuse `Table` component